internal/statemachine/classic_states.go

```diff
@@ -300,6 +304,23 @@ func (stateMachine *StateMachine) calculateStates() error {
 		}
 	}
 
+	// only run makeDisk if there is an artifact to make
+	if classicStateMachine.ImageDef.Artifacts.Qcow2 != nil {
+		// only run make_disk once
+		found := false
+		for _, stateFunc := range rootfsCreationStates {
+			if stateFunc.name == "make_disk" {
+				found = true
+			}
+		}
+		if !found {
+			rootfsCreationStates = append(rootfsCreationStates,
+				stateFunc{"make_disk", (*StateMachine).makeDisk})
```

Comment:
Not a big problem IMO, but will this mean that if someone asks for a `.qcow2` artifact, they'll also get a `.img` automatically? Or is the raw one removed somehow when not needed?

Reply:
Yes, this is the current design. I can add some logic to clean up if you think that would be preferable.

Reply:
Let's add a card in the backlog for that and leave it as is. Since I think we should set certain expectations and not seemingly do more than the user requests, as they might be confused. I'll create the card.